ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -189,72 +262,304 @@ def waypoints_cb(self, lane_msg):
+            curpt.JMT_ptr = -1
+
+    def init_acceleration(self, start_ptr, num_wps):
+        # starts the car moving with a lead foot
+        offset = 0
+
+        velocity = max(0.4, self.waypoints[start_ptr].get_v())
+        self.waypoints[start_ptr].set_v(velocity)
+        self.waypoints[start_ptr].JMTD.set_VAJt(
+                    velocity, self.default_accel, 0.0, 0.0)
+        offset += 1
+        while velocity < 1.5 and offset < num_wps:
+            disp = self.waypoints[start_ptr + offset].get_s() -\
+                self.waypoints[start_ptr].get_s()
+            velocity = min(max(0.5, math.sqrt(self.default_accel * disp * 2.0)),
+                           self.waypoints[start_ptr + offset].get_maxV())
+            rospy.loginfo("velocity set to {} using accel = {} and "
+                          "disp = {} at ptr = {}"
+                          .format(velocity, self.default_accel, disp,
+                                  start_ptr + offset))
+            self.waypoints[start_ptr + offset].set_v(velocity)
+            self.waypoints[start_ptr + offset].JMTD.set_VAJt(
+                velocity, self.default_accel, 0.0, 0.0)
+            self.waypoints[start_ptr + offset].JMT_ptr = -1
+            offset += 1
+        return offset - 1
+
+    def generate_speedup(self, start_ptr, end_ptr):
+        # this is triggered after initiate_aceleration()
+        # gets the car moving - that way we can adjust them
+        # independently
+        recalc = False
+        curpt = self.waypoints[start_ptr]
+
+        if curpt.JMT_ptr == -1 or self.state != 'speedup':
+            self.state = 'speedup'
+            curpt.JMT_ptr = self.setup_jmt(curpt, self.default_velocity)
+            # smooth_factor = 1.0, time_factor = 1.0)
+            jmt_ptr = curpt.JMT_ptr
+        else:
+            jmt_ptr = curpt.JMT_ptr
+            rospy.loginfo("using old jmt_ptr = {}".format(curpt.JMT_ptr))
+        JMT_instance = self.JMT_List[jmt_ptr]
+
+        t = 0.0
+        for ptr in range(start_ptr+1, end_ptr):
+            mod_ptr = ptr % len(self.waypoints)
+            curpt = self.waypoints[mod_ptr]
+
+            if curpt.get_s() <= JMT_instance.final_displacement:
+                # create the main part of the jmt curve
+                t = self.gen_point_in_jmt_curve(curpt, jmt_ptr, t)
+                if self.check_point(curpt) is True:
+                    recalc = True
+            else:
+                # The car has reached the point where it is near to target
+                # velocity
+                rospy.loginfo("{} beyond S = {} at ptr_id = {}".format
+                              (curpt.get_s(), JMT_instance.final_displacement,
+                               mod_ptr))
+                self.maintain_speed(mod_ptr, 1)
+        return recalc
+
+    def set_waypoints_velocity(self):
+        offset = 0           # offset in front of car to account for latency
+        safety_factor = 2.0  # increase slowing down distance
```

Comment:
I would rename "safety_factor" to "accel_ratio" or something to be more clear.  Same with its use as "smooth_factor" on line 515.